consensus/transaction_validator/transaction_validator.c

```diff
@@ -23,14 +23,13 @@ static bool has_invalid_timestamp(transaction_validator_t* const tv,
   uint64_t max_future_timestamp_ms =
       current_timestamp_ms() + MAX_TIMESTAMP_FUTURE_MS;
   if (transaction->attachment_timestamp == 0) {
-    return (transaction->timestamp < tv->snapshot_timestamp_ms &&
+    return (transaction->timestamp * 1000 < tv->snapshot_timestamp_ms &&
```

Comment:
Nits: maybe have a `tv->snapshot_timestamp` too to avoid repetitive useless computation of the timestamp ?

Reply:
I prefer having one field only